### PR TITLE
Add unit tests for `calculate_funding_fee`

### PR DESCRIPTION
**Pull request by luckysori**

### CODE REVIEW

File: daemon/src/model.rs [hunk 1]
```diff
@@ -1275,4 +1276,47 @@ mod tests {
 
         assert_eq!(balance, SignedAmount::from_sat(1000))
     }
+
+    #[test]
+    fn proportional_funding_fees_if_sign_of_funding_rate_changes() {
+        let price = Price::new(Decimal::from_u32(35_000).unwrap()).unwrap();
```
Comment (thomaseizinger) [inline daemon/src/model.rs]: Why not use the `dec!` macro?

File: daemon/src/model.rs [hunk 2]
```diff
@@ -1275,4 +1276,47 @@ mod tests {
...
+        assert_eq!(
+            long_pays_short_fee.fee,
+            short_pays_long_fee.fee * long_leverage.get().into()
+        )
+    }
+
+    #[test]
+    fn zero_funding_fee_if_zero_funding_rate() {
+        let fee = calculate_funding_fee(
+            Price::new(Decimal::from_u32(35_000).unwrap()).unwrap(),
```
Comment (thomaseizinger) [inline daemon/src/model.rs]: Perhaps use `dummy_price` if you don't care?

File: daemon/src/model.rs [hunk 3]
```diff
@@ -1275,4 +1276,47 @@ mod tests {
...
+            Usd::new(Decimal::from_u32(100).unwrap()),
+            Leverage::new(2).unwrap(),
+            FundingRate::new(Decimal::ZERO).unwrap(),
```
Comment (thomaseizinger) [inline daemon/src/model.rs]: Extract into variable and make it the "arrange" part to make it more obvious what the influential variable is in the test?

**Review by thomaseizinger — APPROVED** [review]
Some ideas!

File: daemon/src/model.rs (hunk 1 above)
Comment (luckysori) [reply daemon/src/model.rs]: I forgot its existence!

**Review by thomaseizinger — APPROVED** [review]

File: daemon/src/model.rs [hunk 4]
```diff
@@ -1275,4 +1275,62 @@ mod tests {
...
+        let epsilon = (long_pays_short_fee.fee.as_sat() as i64)
+            - (short_pays_long_fee.fee.as_sat() as i64) * (long_leverage.get() as i64);
+        assert!(epsilon.abs() < 5)
+    }
+
+    #[test]
+    fn zero_funding_fee_if_zero_funding_rate() {
+        let zero_funding_rate = FundingRate::new(Decimal::ZERO).unwrap();
+
+        let dummy_leverage = Leverage::new(1).unwrap();
```
Comment (thomaseizinger) [inline daemon/src/model.rs:1312]: We might want to introduce constants for this at some point.

Comment (luckysori) [reply daemon/src/model.rs:1312]: Do you mean a crate-wide `DUMMY_LEVERAGE` constant, for example?

Comment (thomaseizinger) [reply daemon/src/model.rs:1312]: Uhm, no I was thinking of just constants like `Leverage::ONE` etc

Comment (luckysori) [reply daemon/src/model.rs:1312]: Okay, that makes more sense. But the purpose would not be for tests specifically then, right?

**Pull request merged**